### PR TITLE
STM32: UF2 build option for STM32F405

### PR DESCRIPTION
**Pull request by hierophect**
This PR adds a build option to compile the Feather F405 for use with a UF2 bootloader, such as https://github.com/adafruit/uf2-stm32f. This setting can be enabled for other STM32 boards in the future, though it's not recommended to use a UF2 bootloader on any board without external flash. This PR also makes some minor style changes to the Makefile and removes some unused variables.

### CODE REVIEW

File: ports/stm32f4/boards/feather_stm32f405_express/mpconfigboard.mk [hunk 1]
```diff
@@ -14,6 +14,8 @@ MCU_VARIANT = stm32f4
 MCU_SUB_VARIANT = stm32f405xx
 MCU_PACKAGE = 64
 CMSIS_MCU = STM32F405xx
-LD_FILE = boards/STM32F405.ld
-TEXT0_ADDR = 0x08000000
-TEXT1_ADDR = 0x08010000
\ No newline at end of file
+
+LD_FS = boards/STM32F405_fs.ld # Default to internal FS
```
Comment (tannewt) [inline ports/stm32f4/boards/feather_stm32f405_express/mpconfigboard.mk]: Why internal FS? Doesn't the feather have an external SPI flash?

File: ports/stm32f4/boards/STM32F405_boot.ld [hunk 2]
```diff
@@ -1,39 +1,30 @@
...
 
 /* produce a link error if there is not this amount of RAM for these sections */
 _minimum_stack_size = 2K;
 _minimum_heap_size = 16K;
 
-/* Define the stack.  The stack is full descending so begins just above last byte
-   of RAM.  Note that EABI requires the stack to be 8-byte aligned for a call. */
-_estack = ORIGIN(RAM) + LENGTH(RAM) - _estack_reserve;
-_sstack = _estack - 16K; /* tunable */
+/* Define tho top end of the stack.  The stack is full descending so begins just
```
Comment (tannewt) [inline ports/stm32f4/boards/STM32F405_boot.ld]:
```suggestion
/* Define the top end of the stack.  The stack is full descending so begins just
```

File: ports/stm32f4/boards/feather_stm32f405_express/mpconfigboard.mk (hunk 1 above)
Comment (hierophect) [reply ports/stm32f4/boards/feather_stm32f405_express/mpconfigboard.mk]: It does. This just defines an internal filesystem anyway as a default, the way micropython does. Wastes 48K of flash, but if the user cuts the external flash, it'll allow the board to work. Do you want it cut, or a third option added?

Comment (tannewt) [reply ports/stm32f4/boards/feather_stm32f405_express/mpconfigboard.mk]: I would prefer to use the 48k for CircuitPython flash. We don't support removing external flash on any other board.

**Comment by tannewt** [comment]
This ready for another look?

**Comment by hierophect** [comment]
> This ready for another look?

Yes, it didn't give me the prompt to ask for your re-review. It's all set.

**Comment by tannewt** [comment]
@hierophect Please always add a comment mentioning me. They are easiest to see.

**Comment by hierophect** [comment]
@hathach the `_usbd_dev` object is empty - where should I poke around to try and find the root cause of that? 
```
p _usbd_dev
$2 = {{connected = 0 '\000', configured = 0 '\000', suspended = 0 '\000', remote_wakeup_en = 0 '\000', remote_wakeup_support = 0 '\000', self_powered = 0 '\000'}, itf2drv = '\377' <repeats 16 times>, 
  ep2drv = {"\377\377", "\377\377", "\377\377", "\377\377", "\377\377", "\377\377", "\377\377", "\377\377"}, ep_status = {{{busy = false, stalled = false}, {busy = false, stalled = false}}, {{
        busy = false, stalled = false}, {busy = false, stalled = false}}, {{busy = false, stalled = false}, {busy = false, stalled = false}}, {{busy = false, stalled = false}, {busy = false, 
        stalled = false}}, {{busy = false, stalled = false}, {busy = false, stalled = false}}, {{busy = false, stalled = false}, {busy = false, stalled = false}}, {{busy = false, stalled = false}, {
        busy = false, stalled = false}}, {{busy = false, stalled = false}, {busy = false, stalled = false}}}}
```

**Comment by wildestpixel** [comment]
Is it now therefore possible to use a built uf2 on feather_stm32f405_express - you'd still pull boot0 up I assume but how would you flash this ?

**Comment by hierophect** [comment]
There are two kinds of bootloader for the feather - the built in one, which is stored in a special region of memory, and the UF2 one, which is stored in the beginning of the flash memory and uploads to the flash after itself. To use the UF2 bootloader, you'll need to actually program it into flash using either the built-in bootloader, or by using a dedicated programmer over SWD or JTAG. 

The UF2 bootloader is a separate binary that you need to compile from here: https://github.com/adafruit/uf2-stm32f. Once you have the .bin file, program it onto the feather with DFU util. Then, to create bootloader-friendly UF2 files for Circuitpython, build Circuitpython with the flag UF2_BOOTLOADER, ie `make BOARD=feather_stm32f405_express UF2_BOOTLOADER=1`. 

The UF2 bootloader will appear as a disk image every other reset. You DO NOT use the boot0 pin, that only affects the DFU bootloader and pulling it up would put it into DFU bootloader mode, which you don't want.

**Comment by wildestpixel** [comment]
All working and Beta 5.40 on the board as UF2 using `make BOARD=feather_f405`.

Only small thing is when in bootloader mode board shows as 

Capture

But fully functioning - many thanks hierophect 👍

**Comment by hierophect** [comment]
Yes, we haven't gotten around to changing it to exactly match the Adafruit UF2 bootloaders. I'm glad it went smoothly for you!

**Pull request merged**